### PR TITLE
Introduce journal for formatted output

### LINKED ISSUE #124: Format console output
Currently the console output is a mess, which needs to be cleaned up.

### PR DESCRIPTION
**Pull request by mschreter**
This PR introduces at least a minimal formatting of the output. The implementation is not perfect, however, the output is now much clearer than the old one.

Note: This PR is a rather delicate one, since every output file has changed. I re-created the output-files in release mode. Please take a look at it carefully @peterrum @nmuch.

closes #124

### CODE REVIEW

File: source/melt_pool/melt_pool_problem.cpp [hunk 1]
```diff
@@ -367,18 +374,17 @@ namespace MeltPoolDG::Flow
...
           if (heat_operation)
-            scratch_data->get_pcout() << " T.size " << heat_operation->get_temperature().size();
+            str << " T.size " << heat_operation->get_temperature().size();
           if (melt_pool_operation)
-            scratch_data->get_pcout() << " solid.size " << melt_pool_operation->get_solid().size();
-
-          scratch_data->get_pcout() << std::endl;
+            str << " solid.size " << melt_pool_operation->get_solid().size();
 
+          Journal::print_line(scratch_data->get_pcout(), str.str(), "melt_pool_problem");
```
Comment (peterrum) [inline source/melt_pool/melt_pool_problem.cpp:387]: This looks odd with the additional `std::ostringstream`. But the result looks good!

**Review by peterrum — APPROVED** [review]

**Comment by peterrum** [comment]
> Note: This PR is a rather delicate one, since every output file has changed.

The changes should be fine. As far as I see it, you have not changed any functionality. Except maybe that in the `TimeIterator` `old_time` is saved.

**Comment by peterrum** [comment]
`build-adaflo-release-9-3` has passed :+1:

**Review by peterrum — APPROVED** [review]

File: source/normal_vector/normal_vector_operation.cpp [hunk 2]
```diff
@@ -65,19 +66,23 @@ namespace MeltPoolDG::NormalVector
...
+        MeltPoolDG::VectorTools::compute_L2_norm<dim>(
+          solution_normal_vector.block(d), *scratch_data, normal_dof_idx, normal_quad_idx),
+        "normal_" + std::to_string(d),
+        "normal_vector",
+        11 /*precision*/
+      );
+
+    Journal::print_line(scratch_data->get_pcout(1),
+                        "     * CG: i = " + std::to_string(iter),
+                        "normal_vector");
```
[the remark this answers is not in the document]
Comment (mschreter) [reply source/normal_vector/normal_vector_operation.cpp:85]: Yes :-). At some point we could store the "verbosity_level::test" somewhere and restore it via `scratch_data->get_pcout(verbosity_level::test)`.

File: source/melt_pool/melt_pool_problem.cpp (hunk 1 above)
Comment (mschreter) [reply source/melt_pool/melt_pool_problem.cpp:387]: Yes that's true. I've tried to spend not too much time on the really unhandy formatting of C++.

**Comment by mschreter** [comment]
> > Note: This PR is a rather delicate one, since every output file has changed.
> 
> The changes should be fine. As far as I see it, you have not changed any functionality. Except maybe that in the `TimeIterator` `old_time` is saved.

That's correct. Thanks a lot Peter for having a look at it :-).

**Comment by peterrum** [comment]
For sake of completeness. These are the test that are failing on my machine:
```
The following tests FAILED:
	 37 - tests/solidification_slab.debug (Failed)
	 53 - tests/unidirectional_heat_transfer_with_two_phase.mpirun=4.debug (Failed)
	 60 - tests_not_robust/recoil_pressure_amr.mpirun=4.debug (Failed)
	 62 - tests_not_robust/stefans_problem2_with_flow_and_heat.mpirun=4.debug (Failed)

```

**Pull request merged**